### PR TITLE
[CELEBORN-1125] Bump guava from 14.0.1 to 32.1.3-jre

### PR DESCRIPTION
**Pull request by cfmcgrady**
### What changes were proposed in this pull request?

As title

### Why are the changes needed?

- bump guava from 14.0.1 to 32.1.3-jre
- refer to https://github.com/apache/spark/pull/26911, remove usages of Guava that no longer work in Guava 27/32, and replace with workalikes. After this PR, Celeborn no longer relies on a specific version of Guava, and is compatible with Guava 14/27/32. we have the ability to specify Guava to 27 when running MapReduce integration tests.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Pass GA

### CODE REVIEW

File: common/src/main/java/org/apache/celeborn/common/network/buffer/FileSegmentManagedBuffer.java [hunk 1]
```diff
@@ -145,10 +146,10 @@ public long getLength() {
 
   @Override
   public String toString() {
```
Comment (cfmcgrady) [inline common/src/main/java/org/apache/celeborn/common/network/buffer/FileSegmentManagedBuffer.java:148]: before
```
FileSegmentManagedBuffer{file=/tmp/file, offset=1, length=1}
```

after
```
FileSegmentManagedBuffer[file=/tmp/file,offset=1,length=1]
```

Comment (FMX) [reply common/src/main/java/org/apache/celeborn/common/network/buffer/FileSegmentManagedBuffer.java:148]: I think this change is acceptable.

**Review by FMX — APPROVED** [review]
LGTM.

**Comment by FMX** [comment]
Merged into main.

File: project/CelebornBuild.scala [hunk 2]
```diff
@@ -72,7 +72,12 @@ object Dependencies {
   val commonsLang3 = "org.apache.commons" % "commons-lang3" % commonsLang3Version
   val commonsLogging = "commons-logging" % "commons-logging" % commonsLoggingVersion
   val findbugsJsr305 = "com.google.code.findbugs" % "jsr305" % findbugsVersion
-  val guava = "com.google.guava" % "guava" % guavaVersion
+  val guava = "com.google.guava" % "guava" % guavaVersion excludeAll(
+    ExclusionRule("org.checkerframework", "checker-qual"),
+    ExclusionRule("org.codehaus.mojo", "animal-sniffer-annotations"),
+    ExclusionRule("com.google.errorprone", "error_prone_annotations"),
+    ExclusionRule("com.google.guava", "listenablefuture"),
```
Comment (pan3793) [inline project/CelebornBuild.scala:79]: is this one a runtime dep?

**Pull request closed without merging**